HyphaApp/hypha
# Afford Submission visibility settings per Group

Administrators can see "Related submissions" on a Submissions Details page. This change allows a Wagtail administrator to add visibility to any Group such that members of the Group can also see "Related submissions." This change embraces Django Group and Wagtail Setting Models such that the settings can be made within the application.

A Wagtail administrator can add visibility from the `admin` interface via Settings -> Submission Details Page Settings. To grant visibility, the administrator checks the "Sees related submissions" box for the Group specified in the Group dropdown. To revoke visibility, the administrator unchecks the same. However, if a user is a member of any group that has been granted visibility, the user will have visibility.

Fixes #3958

## Test Steps

Using the sandbox database.

- Log in as Reviewer
- Go to submission 2 (`apply/submissions/2/`)
- Verify that the "Related submissions" box is not present.
- Log in as Admin
- Go to submission 2 
- Verify that the "Related submissions" box is present.
- Go to Wagtail admin (`/admin`)
- Go to "Settings" -> "Submissions Details Page Settings" (`admin/settings/apply_utils/submissionsdetailssetting/`)
- Add a row for "Contracting" and check the "Sees related submissions box."
- Log in as Reviewer again
- Go to submission 2 again
- Verify that the "Related submissions" box is present.
